### PR TITLE
test containers support for tests #368

### PR DESCRIPTION
**Pull request by nikoncode**
@vladmihalcea please review my changes for 52 before I will replicate it for all packages.

I have 1 PG test failed, but it also failed without my changes (using docker-compose). (most likely it's environmental issue with timezones)

```
com.vladmihalcea.hibernate.type.range.guava.PostgreSQLGuavaRangeTypeTest#test
java.lang.AssertionError: 
Expected :[2007-05-03T10:15:30+01:00..2008-05-03T10:15:30+01:00]
Actual   :[2007-05-03T11:15:30+02:00..2008-05-03T11:15:30+02:00]
```

### CODE REVIEW

**Comment by vladmihalcea** [comment]
Thanks, I'll review it when I have some time. Unfortunately, I don't have too much time at the moment.

File: hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/util/providers/DataSourceProvider.java [hunk 1]
```diff
@@ -12,20 +12,5 @@
 
     DataSource dataSource();
```
Comment (vladmihalcea) [inline hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/util/providers/DataSourceProvider.java:14]: I wouldn't delete those methods as they could be useful. These are JDBC client-specific settings, so they should not interfere with this feature.

File: hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/util/providers/H2DataSourceProvider.java [hunk 2]
```diff
@@ -19,41 +17,12 @@ public String hibernateDialect() {
     @Override
     public DataSource dataSource() {
         return JdbcConnectionPool.create(
-            url(),
```
Comment (vladmihalcea) [inline hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/util/providers/H2DataSourceProvider.java:22]: It's better to keep the existing setting as some of the existing committers might use them.

File: hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/util/providers/MySQLDataSourceProvider.java [hunk 3]
```diff
@@ -1,73 +1,23 @@
...
 
 /**
  * @author Vlad Mihalcea
  */
 public class MySQLDataSourceProvider implements DataSourceProvider {
+    private static final MySQLContainer<?> MYSQL_CONTAINER;
 
-    private boolean rewriteBatchedStatements = true;
+    static {
+        MYSQL_CONTAINER = new MySQLContainer<>("mysql:5.7")
```
Comment (vladmihalcea) [inline hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/util/providers/MySQLDataSourceProvider.java:15]: I wouldn't start the container unless we cannot find the DB using the default settings. So, I'd use lazy initialization for it at bootstrap time. If we get a connection failure, then we bootstrap the Testcontainers DB.

**Comment by nikoncode** [comment]
@vladmihalcea if I will do initialization in the constructor, it will start/stop container every test class, I don't believe we expect that. Especially when we are running the whole suit.

Now, I see two options:
* since I know the default url, username, password I can implement a check on "static" level before running container. If I can connect using default credentials, then skip test containers. Otherwise, it will spin the container once.
* taking into account that you don't want test containers to be default behaviour maybe it's better for me to just actualize docker-compose and save efforts for the next contribution?

**Comment by vladmihalcea** [comment]
The docker-compose option is also a good option.

**Comment by vladmihalcea** [comment]
Support for Testcontainers was added by [this commit](https://github.com/vladmihalcea/hibernate-types/commit/a571dbe03c04991361ace4eaeaeeee2352e81921). Thanks for your PR.

**Pull request closed without merging**